### ISSUE TITLE
Error of no DPU core found in current configuration of DPU IP!

### ISSUE THREAD

**Issue by Pranjali-Shinde** [Xilinx/Edge-AI-Platform-Tutorials#26]
Hi,
I am following the DPU integration tutorial. I am getting this error when I boot on ZU4EV board. Although, I have checked for the previous answers on the Xilinx forum none of solutions did apply on solving this error. The error presented is shown below.

[    4.121067] [DPU][2102]Found DPU signature addr = 0xbf000000 in device-tree
[    4.128023] [DPU][2102]Checking DPU signature at addr = 0xbff00000,
[    4.134387] [DPU][2102]DPU signature NOT found, fallback to device-tree.
[    4.141082] [DPU][2102]Error of no DPU core found in current configuration of DPU IP!

In addition, when the DPU signature and signature VA is checked by printing the fields in dpucore.c file (line 1139)

if (signature_addr != SIG_BASE_NULL) {
dpr_init("Checking DPU signature at addr = 0x%lx, \n", signature_addr);
signature_va =
ioremap((phys_addr_t)signature_addr, 1 * sizeof(signature_field));
signature_field = ioread32(signature_va);

dpr_init("Checking DPU signature field is = 0x%lx, \n",signature_field);
dpr_init("Checking DPU signature VA field is = 0x%lx, \n",signature_va);

I get the following error on booting the board
[system_user_dtsi.txt](https://github.com/Xilinx/Edge-AI-Platform-Tutorials/files/3502351/system_user_dtsi.txt)
:

root@FM191_kernel:~# lsmod
Tainted: G
dpu 49152 0 - Live 0xffffff8000b23000 (O)
mali 241664 0 - Live 0xffffff8000ad0000 (O)
uio_pdrv_genirq 16384 0 - Live 0xffffff8000ac0000

root@FM191_kernel:~# dmesg

[ 4.340785] [DPU][2105]Found DPU signature addr = 0xbf000000 in device-tree
[ 4.347747] [DPU][2105]Checking DPU signature at addr = 0xbff00000,
[ 4.354115] [DPU][2105]Checking DPU signature field is = 0x0,
[ 4.359941] [DPU][2105]Checking DPU signature VA field is = 0xffffff80093ad000,
[ 4.367327] [DPU][2105]DPU signature NOT found, fallback to device-tree.
[ 4.374025] [DPU][2105]Error of no DPU core found in current configuration of DPU IP!
[ 4.393505] EXT4-fs (mmcblk1p2): re-mounted. Opts: data=ordered
[ 4.694489] pps pps0: new PPS source ptp0
[ 4.698515] macb ff0e0000.ethernet: gem-ptp-timer ptp clock registered.
[ 4.705196] IPv6: ADDRCONF(NETDEV_UP): eth0: link is not ready
[ 4.940190] PLL: shutdown
[ 7.824223] macb ff0e0000.ethernet eth0: link up (1000/Full)
[ 7.829904] IPv6: ADDRCONF(NETDEV_CHANGE): eth0: link becomes ready

I attaching the system-user.dtsi.

The PL interface is PL_PS_IRQ1 the GIC RIQ number is 143:136. The interrupt number calculated is (136-32=104) which is 68 hex.

**Comment by shengwenLeong** [comment]
May be your address configure of DPU is incorrect. Could you show the address configuration of DPU IP in vivado?

**Comment by Pranjali-Shinde** [comment]
Yes. Please find the attachment. When I run the command report_property [get_bd_cells dpu_eu_0] and when I check the value of the following is:

CONFIG.S_AXI_SLAVES_BASE_ADDR 0x80000000
whereas on the address editor it shows as 0xBF000000

![vivado_address_editor](https://user-images.githubusercontent.com/26780627/64245147-4c63bd80-cf02-11e9-9fc0-99e1b79a1af6.png)

**Comment by shengwenLeong** [comment]
I check the system_user_dtsi.txt file and address configuration, but i did not find any problem. I have run the DPU successfully on the Zynq 7045 chip according to the [guide](https://github.com/sumilao/Zynq-7000-DPU-TRD). When I first run DPU, i also face the problem like you.
```
[ 4.121067] [DPU][2102]Found DPU signature addr = 0xbf000000 in device-tree
[ 4.128023] [DPU][2102]Checking DPU signature at addr = 0xbff00000. 
```
Then, I find my vivado project has a problem, the DPU is not configured correctly. So maybe you should check the DPU configuration. And this project uses DPU is v2.0 and DPU driver only for DPU v2.0. If you use DPU v3.0, you should use the dpu.ko provided by the Xilinx with Petalinux 2019.1 (important). The DPU driver in this project unsuits for DPU v3.0.

**Comment by pedrombmachado** [comment]
Both,
  You are trying to generate the boot files with the wrong parameters. Compile the project in Vivado and then generate the FSBL using the XSDK. There will be an autogenerated file called xparameters.h that contains the base addresses, IRQs and type of interruption. After that you just have to use that info for generating the device tree.

**Comment by yunqu** [comment]
I have seen similar things. I did my own bitstream based on the DPU tutorial. The petalinux bsp is based on official released zcu104 bsp (not dpu one), and I made device tree correctly include the dpu segment. dpu.ko is built for my rootfs.

```
[  226.532866] [DPU][4588]Found DPU signature addr = 0x8f000000 in device-tree
[  226.539840] [DPU][4588]Checking DPU signature at addr = 0x8ff00000,
[  226.546313] [DPU][4588]Error of no DPU core found in current configuration of DPU IP!
```

Is it really necessary to use the DPU hardware design to build the boot files? Can't I just reload bitstream during run-time?

**Comment by pedrombmachado** [comment]
@yunqu , that is because you have not configured the correct configuration for the DPU.
 Get the base address, irq and irq type from the xparameters.h in the BSP incude files

**Comment by yunqu** [comment]
@pedrombmachado I double checked; the dts and the irq are both correct. I don't think it is the problem of the dts; maybe I really need to use DPU as the starting point of my petalinux bsp? Remember I am just using the official zcu104 bsp, which contains a different hardware design.

**Comment by pedrombmachado** [comment]
@yunqu you have to use the new bsp i.e:
1) create your custom design in Vivado;
2) compile and export the hardware in Vivado
3) Launch sdk from vivado
4) Create the FSBL from the xsdk
5) Check the new xparameters.h in the new bsp.
Have a look in our wiki with the steps for [building firmware](https://github.com/SundanceMultiprocessorTechnology/VCS-1/wiki/Build-Firmware)

**Comment by yunqu** [comment]
@pedrombmachado I rebuilt the bsp starting from a DPU design, and was able to boot up the board, but now the linux hangs when I insmod. It hangs after

```
[  226.532866] [DPU][4588]Found DPU signature addr = 0x8f000000 in device-tree
[  226.539840] [DPU][4588]Checking DPU signature at addr = 0x8ff00000,
```

**Comment by pedrombmachado** [comment]
Hi @yunqu ,
   the xparameters.h has the info that you are asking. Also, I think your `interrupt-parent` may be wrong.
Look how mine looks like.
```
&amba{
        dpu{
                compatible = "xilinx,dpu";
                base-addr = <0x8f000000>;
                dpucore{
                        compatible = "xilinx,dpucore";
                        interrupt-parent = <&gic>;
                        interrupts = <0x0 0x7B 0x1>;
                        core-num = <0x1>;
                };
                softmax{
                        compatible = "xilinx,smfc";
                        interrupt-parent = <&gic>;
                        interrupts = <0x0 0x7F 0x1>;
                        core-num = <0x1>;
                };
        };
};
```

**Comment by yunqu** [comment]
@pedrombmachado I have found out my configuration is correct. The problem is that I need `busybox` in my rootfs in order to access specific memory regions. Since I have a ubuntu OS, after `apt-get install busybox`, I was able to see:
```
[   53.014297] [DPU][4218]Found DPU signature addr = 0x8f000000 in device-tree
[   53.021282] [DPU][4218]Checking DPU signature at addr = 0x8ff00000,
[   53.027764] [DPU][4218]DPU signature checking done!
[   53.033918] [DPU][4218]Init SMFC IP...
[   53.037713] [DPU][4218]Request SMFC IRQ 54 successful.
[   53.037718] [DPU][4218]Init SMFC IP done
```

**Comment by pedrombmachado** [comment]
@yunqu got to know.
  I propose that @Pranjali-Shinde closes this issue because it is now solved!
